src/coreclr/jit/liveness.cpp

```diff
@@ -351,6 +338,50 @@ void Compiler::fgPerNodeLocalVarLiveness(GenTree* tree)
     }
 }
 
+#if defined(FEATURE_HW_INTRINSICS)
+void Compiler::fgPerNodeLocalVarLiveness(GenTreeHWIntrinsic* hwintrinsic)
+{
+    NamedIntrinsic intrinsicId = hwintrinsic->GetHWIntrinsicId();
+
+    // We can't call fgMutateGcHeap unless the block has recorded a MemoryDef
+    //
+    if (hwintrinsic->OperIsMemoryStore())
```

Comment:
```suggestion
    if (hwintrinsic->OperIsMemoryStoreOrBarrier())
```
And delete the `switch` below?